File: Tree/Java/Tree_01.java
public class Tree_01
{
    public static class Node
    {
        int data;
        Node left = null;
        Node right = null;

        Node(int data)
        {
            this.data=data;
        }
    }

    static int idx = 0;
    public static Node construct_Tree(int[] arr)
    {
        if(idx == arr.length || arr[idx] == -1)
        {
            idx++;
            return null;
        }

        Node node = new Node(arr[idx++]);
        node.left = construct_Tree(arr);
        node.right = construct_Tree(arr);

        return node;
    }

    public static void display(Node node)
    {
        if(node == null)
        {
            return;
        }

        String str = "";
        str += ((node.left != null) ? node.left.data : ".");
        str += " <- " + node.data + " -> ";
        str += ((node.right != null) ? node.right.data : ".");
        System.out.println(str);

        display(node.left);
        display(node.right);

    }

    public static void Tree_Basics()
    {
        int[] arr = {10, 20, 40, -1, -1, 50, 80, -1, -1, 90, -1, -1, 30, 60, 100, -1, -1, -1, 70, 110, -1, -1, 120, -1, -1};
        Node root = construct_Tree(arr);
        display(root);
    }
    public static void solve()
    {
        Tree_Basics();
    }
    public static void main(String[] args)
    {
        solve();
    }
}

File: Tree/CPP/Tree_01.cpp
#include<iostream>
#include<vector>

using namespace std;

class Node
{

public:
    
    int data;
    Node* left=nullptr;
    Node* right=nullptr;

    Node(int data)
    {
        this->data = data;
    }
    
};

int idx = 0;
Node* construct_Tree(vector<int>& arr)
{
    if(idx == arr.size() || arr[idx] == -1)
    {
        idx++;
        return nullptr; //NULL
    }

    Node* node = new Node(arr[idx++]);
    node->left = construct_Tree(arr);
    node->right = construct_Tree(arr);

    return node;
}

void display(Node* node)
{
    if(node == nullptr)
    {
        return;
    }

    string str = "";
    str += ((node->left != nullptr) ? to_string(node->left->data) : ".");
    str += " <- " + to_string(node->data) + " -> ";
    str += ((node->right != nullptr) ? to_string(node->right->data) : ".");
    cout << str << endl;

    display(node->left);
    display(node->right);

}

void Tree_Basics()
{
    vector<int> arr = {10, 20, 40, -1, -1, 50, 80, -1, -1, 90, -1, -1, 30, 60, 100, -1, -1, -1, 70, 110, -1, -1, 120, -1, -1};
    Node* root = construct_Tree(arr);
    display(root);
}
void solve()
{
    Tree_Basics();
}
int main()
{
    solve();
    return 0;
}